airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -126,38 +127,45 @@ def path(self, **kwargs) -> str:
 
 
 # Incremental Streams
-def chunk_date_range(start_date: DateTime, interval=1) -> Iterable[Mapping[str, any]]:
+def chunk_date_range(start_date: DateTime, interval=pendulum.duration(days=1)) -> Iterable[Period]:
     """
-    Returns a list of the beginning and ending timetsamps of each day between the start date and now.
-    The return value is a list of dicts {'oldest': float, 'latest': float} which can be used directly with the Slack API
+    Yields a list of the beginning and ending timestamps of each day between the start date and now.
+    The return value is a pendulum.period
     """
-    intervals = []
     now = pendulum.now()
     # Each stream_slice contains the beginning and ending timestamp for a 24 hour period
     while start_date <= now:
-        end = start_date.add(days=interval)
-        intervals.append({"oldest": start_date.timestamp(), "latest": end.timestamp()})
-        start_date = start_date.add(days=1)
-
-    return intervals
+        end_date = start_date + interval
+        yield pendulum.period(start_date, end_date)
+        start_date = end_date
 
 
 class IncrementalMessageStream(SlackStream, ABC):
     data_field = "messages"
     cursor_field = "ts"
+    primary_key = "id"
 
     def __init__(self, default_start_date: DateTime, **kwargs):
-        self._default_start_date = default_start_date
+        self._start_ts = default_start_date.timestamp()
         super().__init__(**kwargs)
 
     def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
         params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, **kwargs)
         params.update(**stream_slice)
         return params
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            record[self.primary_key] = record[self.cursor_field]
```

Comment:
To stay in line with ELT, wherever possible we want to keep the schema as close to the source as we can. This allows the user to look up the schema of the Slack API directly and understand what the fields mean. But if we add an undocumented field that is airbyte specific then we have to maintain docs explaining what it is. Also there is a chance of getting this wrong due to manual error (here it's pretty simple but just speaking about the general case).

Reply:
@sherifnada 
the problem is that SlackAPI use ts as a PK, it is formatted as a string because all digits are significant - really "exotic" desigion, 
"12345.230000" will not match 12345.23
we use state to:
-  get datetime df and read replies in period T1 -> dt -> T2 (need to parse timestamp)
-  get id to request replies (need to provide string with 6 digits after dot)
-  during SAT incremental test to compare state with cursor in records

We have few options:
- use floats and serialize them to string (really bad, as we don't know if the format will change)
- use text as it is this means we need hack/extend SAT to support only this weird case (can't compare numbers in strings '3' > '12')
- use primary_key column as it is, add cursor_column (ts) at save there `float` value

Reply:
These are good points, thanks for the clarification Eugene. In this case can we keep `ts` unedited and instead of `id` use `ts_float`? this way we don't alter the original schema

Reply:
Done